src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -861,6 +875,11 @@ private void ProcessSettingsFrame(FrameHeader frameHeader, bool initialFrame = f
                             }
                             break;
 
+                        case SettingId.MaxHeaderListSize:
+                            _maxHeaderListSize = settingValue;
+                            _pool._largestSeenHttp2MaxHeaderListSize = Math.Max(_pool._largestSeenHttp2MaxHeaderListSize, _maxHeaderListSize);
```

Comment:
Do you think we shouldn't worry of the multi-threaded access (in case of multiple H2 connections) here?

Reply:
It shouldn't matter either way as this is only a best-effort guess until the server sends us the actual value. But I guess it's fewer LOC to just add the lock than for a comment on why it may not actually matter in practice, I'll add it